src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -466,6 +466,20 @@ case class TagFamilySizeMetric(family_size: Int,
                                var fraction: Proportion = 0,
                                var fraction_gt_or_eq_family_size: Proportion = 0) extends Metric
 
+/**
+ * Metrics produced by `GroupReadsByUmi` to describe reads passed through UMI grouping
+ * @param reads The number of reads accepted for grouping
+ * @param filteredNonPf The number of non-PF reads
+ * @param filteredPoorAlignment The number of templates discarded for poor
+ * @param filteredNsInUmi The number of templates discarded due to one or more Ns in the UMI
+ * @param filterUmisTooShort The number of templates discarded due to a shorter than expected UMI
+ */
+case class UmiGroupingMetric(reads: Long,
+                             filteredNonPf: Long,
+                             filteredPoorAlignment: Long,
+                             filteredNsInUmi: Long,
+                             filterUmisTooShort: Long) extends Metric
```

Comment:
Metrics are the style exception and should be `snake_case`:

```suggestion
case class UmiGroupingMetric(reads: Long,
                             filtered_non_pf: Long,
                             filtered_poor_alignment: Long,
                             filtered_ns_in_umi: Long,
                             filter_umis_too_short: Long) extends Metric
```